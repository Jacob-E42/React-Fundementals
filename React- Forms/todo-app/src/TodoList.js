import React, { useState } from "react";
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";

const TodoList = () => {
	const [todos, setTodos] = useState([]);
	const add = todo => {
		setTodos(todos => [...todos, todo]);
	};
	const remove = id => {
		setTodos(todos => todos.filter(todo => todo.id !== id));
	};

	const todoComponents = todos.map(todo => {
		return (
			<Todo
				task={todo.task}
				key={todo.id}
				id={todo.id}
				remove={remove}
			/>
		);
	});

	return (
		<div>
			<NewTodoForm add={add} />
			<ul>{todoComponents}</ul>
		</div>
	);
};

export default TodoList;
