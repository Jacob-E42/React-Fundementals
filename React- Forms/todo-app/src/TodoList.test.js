import { render, screen, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

test("smoketest and snapshot test for TodoList", () => {
	const { asFragment } = render(<TodoList />);
	expect(asFragment).toMatchSnapshot();
});

test("form submits new todo properly", () => {
	const { getByLabelText, queryByText } = render(<TodoList />);
	const taskInput = getByLabelText("Task");
	const button = queryByText("Submit");

	expect(queryByText("go to sleep")).not.toBeInTheDocument();
	fireEvent.change(taskInput, { target: { value: "go to sleep" } });
	fireEvent.click(button);

	expect(queryByText("go to sleep")).toBeInTheDocument();
});
