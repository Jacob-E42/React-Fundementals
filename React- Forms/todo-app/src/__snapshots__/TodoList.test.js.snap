// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoketest and snapshot test for TodoList 1`] = `[Function]`;
